docs/HowToFAQ/FAQThisFrameField.md

```diff
@@ -0,0 +1,25 @@
+---
+title: How do I say 'reads if x then this\`y else this\`z'? Dafny complains about the 'this'.
+---
+
+## Question: 
+
+How do I say 'reads if x then this\`y else this\`z'? Dafny complains about the 'this'.
+
+## Answer:
+
+Here is some sample code that show a workaround.
+
+```dafny
+trait Test {
+  var y: int
+  var z: int
+  predicate readsY() reads this`y { true }
+  predicate readsZ() reads this`z { true }
+  function {:opaque} MyFunc(x: bool): int
+    reads if x then readsY.reads() else readsZ.reads() {
+    if x then y else z
+  }
+}
```

Comment:
@RustanLeino told me that actually the function reads() itself overapproximate in this case :-(
Fortunately, there is another workaround.

```suggestion
trait Test {
  var y: int
  var z: int
  function {:opaque} MyFunc(x: bool): int
    reads (if x then {this} else {})`y, (if x then {} else {this})`z {
    if x then y else z
  }
}
```